apps/nextjs-app-router/src/middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { match } from '@formatjs/intl-localematcher'
+import Negotiator from 'negotiator'
+import { defaultLocale, locales } from '@/localization'
+
+function getLocale(request: NextRequest) {
```

Comment:
Should we drop this function in `src/localization` or is the logic middleware-specific?